examples/seal/seal_policy_auth.c

```diff
@@ -0,0 +1,347 @@
+/* seal_policy_auth.c
+ *
+ * Copyright (C) 2006-2023 wolfSSL Inc.
+ *
+ * This file is part of wolfTPM.
+ *
+ * wolfTPM is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfTPM is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/* This is a helper tool for setting policies on a TPM 2.0 PCR */
+
+#include <wolftpm/tpm2_wrap.h>
+
+#ifndef WOLFTPM2_NO_WRAPPER
+
+#ifndef WOLFTPM2_NO_WOLFCRYPT
+#include <wolfssl/wolfcrypt/hash.h>
+#endif
+
+#include <examples/seal/seal.h>
+#include <hal/tpm_io.h>
+#include <examples/tpm_test.h>
+#include <examples/tpm_test_keys.h>
+
+#include <stdio.h>
+
+/******************************************************************************/
+/* --- BEGIN TPM2.0 PCR Policy example tool  -- */
+/******************************************************************************/
+
+static void usage(void)
+{
+    printf("Expected usage:\n");
+    printf("./examples/pcr/policy [-aes/xor] [pcr]\n");
+    printf("* pcr: PCR index between 0-23 (default %d)\n", 16);
+    printf("* -aes/xor: Use Parameter Encryption\n");
+}
+
+int TPM2_PCR_Seal_With_Policy_Auth_Test(void* userCtx, int argc, char *argv[])
+{
+    int i;
+    int rc = -1;
+    WOLFTPM2_DEV dev;
+    WOLFTPM2_KEY storage;
+    WOLFTPM2_SESSION tpmSession;
+    WOLFTPM2_KEYBLOB authKey;
+    WOLFTPM2_KEYBLOB sealBlob;
+    TPMT_PUBLIC authTemplate;
+    TPMT_PUBLIC sealTemplate;
+    /* default to aes since parm encryption is required */
+    TPM_ALG_ID paramEncAlg = TPM_ALG_CFB;
+    TPM_ALG_ID alg = TPM_ALG_RSA;
+    word32 pcrIndex = 16;
+    byte policyDigest[TPM_MAX_DIGEST_SIZE];
+    word32 policyDigestSz = (word32)sizeof(policyDigest);
+    byte policyDigestSig[MAX_RSA_KEY_BYTES];
+    word32 policyDigestSigSz = (word32)sizeof(policyDigestSig);
+    byte badDigest[TPM_MAX_DIGEST_SIZE] = {0};
+    byte badSig[TPM_MAX_DIGEST_SIZE] = {0};
+    word32 badSigSz = (word32)TPM_MAX_DIGEST_SIZE;
+    word32 pcrArray[256];
+    word32 pcrArraySz = 0;
+    const byte nonce[] = {15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0};
+    const byte secret[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15};
+    byte secretOut[16];
```

Comment:
Avoid hard coded..

Reply:
I don't understand why this can't be hard coded, it's an example so I don't see the downside of everything being clear, visible and contained in a single file for the user